File: C++/41-和为S的连续正数序列.cpp
class Solution
{
public:
    int total(int a, int b)
    {
        return (a + b) * (b - a + 1) / 2;
    }
    vector<vector<int>> FindContinuousSequence(int sum)
    {
        int low = 1, high = 2;
        vector<vector<int>> res;
        while (low < high)
        {
            if (total(low, high) == sum)
            {
                vector<int> t;
                for (size_t i = low; i <= high; i++)
                    t.push_back(i);
                res.push_back(t);
                low++;
            }
            else if (total(low, high) < sum)
                high++;
            else
                low++;
        }
        return res;
    }
};

File: Python/41-和为S的连续正数序列.py
# -*- coding:utf-8 -*-
class Solution:
    def sum(self, a, b):
        return (a + b) * (b - a + 1) / 2

    def FindContinuousSequence(self, tsum):
        # write code here
        res = []
        low, high = 1, 2
        while low < high:
            t = self.sum(low, high)
            if t == tsum:
                res.append(list(range(low, high + 1)))
                low += 1
            elif t < tsum:
                high += 1
            else:
                low += 1
        return res